### ISSUE TITLE
Datatable stuck on processing

### ISSUE THREAD

**Issue by ronaldramlan** [yajra/laravel-datatables#2179]
Trying to fetch MySQL data and display it with datatable. The backend laravel data query seems to work fine and JSON data returned was already confirmed as valid JSON by https://jsonlint.com/. However, the frontend datatable got stuck on "Processing...".

Already checked with a bunch of online forums and questions to find a solution with no luck.

Thanks for helping me find the root cause.

#### layouts: sunrise.blade.php (head)
'<pre><code>

    
    <meta name="csrf-token" content="{{ csrf_token() }}">

    <title>{{ config('app.name', 'Laravel') }}</title>

    
    

    
    <link rel="dns-prefetch" href="//fonts.gstatic.com">
    <link href="https://fonts.googleapis.com/css?family=Nunito" rel="stylesheet">

    
    <link href="{{ asset('css/app.css') }}" rel="stylesheet">

    
    <link href="{{ asset('css/jquery.dataTables.min.css') }}" rel="stylesheet">

    
    <script src="{{ asset('js/jquery-3.4.1.min.js') }}"></script>
    <script src="{{ asset('/js/jquery.dataTables.min.js') }}"></script>

    
    <script src="{{ asset('js/sunrise.js') }}"></script>
</code></pre>'

#### layouts: sunrise.blade.php (body)
'<pre><code>

    <div id="app">
        <nav class="navbar navbar-expand-md navbar-light bg-white shadow-sm">
            <div class="container">
                <a class="navbar-brand" href="{{ url('/') }}">
                    {{ config('app.name', 'Laravel') }}
                </a>
                <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="{{ __('Toggle navigation') }}">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <div class="collapse navbar-collapse" id="navbarSupportedContent">
                    
                    <ul class="navbar-nav mr-auto">

                    </ul>

                    
                    <ul class="navbar-nav ml-auto">
                        
                        @guest
                            <li class="nav-item">
                                <a class="nav-link" href="{{ route('login') }}">{{ __('Login') }}</a>
                            </li>
                            @if (Route::has('register'))
                                <li class="nav-item">
                                    <a class="nav-link" href="{{ route('register') }}">{{ __('Register') }}</a>
                                </li>
                            @endif
                        @else
                            <li class="nav-item dropdown">
                                <a id="navbarDropdown" class="nav-link dropdown-toggle" href="#" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" v-pre>
                                    {{ Auth::user()->name }} <span class="caret"></span>
                                </a>

                                <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
                                    <a class="dropdown-item" href="{{ route('logout') }}"
                                       onclick="event.preventDefault();
                                                     document.getElementById('logout-form').submit();">
                                        {{ __('Logout') }}
                                    </a>

                                    <form id="logout-form" action="{{ route('logout') }}" method="POST" style="display: none;">
                                        @csrf
                                    </form>
                                </div>
                            </li>
                        @endguest
                    </ul>
                </div>
            </div>
        </nav>

        <main class="py-4">
            @yield('content')
        </main>
    </div>
</code></pre>'

#### main view: table.blade.php
'<pre><code>
@extends('layouts.sunrise')

@section('content')
<div class="container">

    <div class="row">

        <div class="col-8 offset-2">

            <div class="form-group row">
                <h1>Table Filter</h1>
            </div>

            <div class="form-group row">
                <label for="data-source" class="col-md-4 col-form-label">Select Data Source</label>

                <select id="data-source" type="text" name="data-source" class="form-control @error('data-source') is-invalid @enderror" data-source="data-source" value="{{ old('data-source') }}" autocomplete="data-source" autofocus>
                    <option value="data_meter">Metering</option>
                    <option value="data_alarm">Alarm</option>
                </select>

                @error('data-source')
                    <span class="invalid-feedback" role="alert">
                        <strong>{{ $message }}</strong>
                    </span>
                @enderror

            </div>

            <div class="form-group row">
                <label for="start-date" class="col-md-4 col-form-label">Start Date</label>

                <input id="start-date" type="datetime-local" name="start-date" class="form-control @error('start-date') is-invalid @enderror" start-date="start-date" value="{{ old('start-date') }}" autocomplete="start-date" autofocus>

                @error('start-date')
                    <span class="invalid-feedback" role="alert">
                        <strong>{{ $message }}</strong>
                    </span>
                @enderror

            </div>
            <div class="form-group row">
                <label for="end-date" class="col-md-4 col-form-label">End Date</label>
                <input id="end-date" type="datetime-local" name="end-date" class="form-control @error('end-date') is-invalid @enderror" end-date="end-date" value="{{ old('end-date') }}" autocomplete="end-date" autofocus>
                @error('end-date')
                    <span class="invalid-feedback" role="alert">
                        <strong>{{ $message }}</strong>
                    </span>
                @enderror
            </div>
            <div class="form-group row pt-4">
                <button class="btn btn-primary" id="apply-filter">Apply Filter</button>
            </div>
        </div>
    </div>
    <div class="row mt-5">
        <table class="table table-bordered" id="sunrise-table">
            <thead>
                <tr>
                    <th>Date Time</th>
                    <th>Equipment Name</th>
                    <th>I_Avg (A)</th>
                    <th>V_Avg (V)</th>
                    <th>Power Factor</th>
                    <th>Acc. Energy Active (GWh)</th>
                    <th>Inc. Energy Active (kWh)</th>
                    <th>THDi (%)</th>
                    <th>THDv (%)</th>
                </tr>
            </thead>
        </table>
    </div>
</div>
@endsection
</code></pre>'

#### main controller: SunriseTableController.php
<pre><code>

namespace App\Http\Controllers;

use Illuminate\Http\Request;

use Illuminate\Support\Facades\DB;

use Illuminate\Database\QueryException;

use Yajra\DataTables\DataTables;

class SunriseTableController extends Controller
{
	public function index()
	{

		return view('sunrise.table');
		
	}

    public function fetchTableContent()
    {

        $input['data-source'] = (request()->get('data_source') === null ? 'data_meter' : request()->get('data_source')) ;
        $input['start-date'] = request()->get('start_date');
        $input['end-date'] = request()->get('end_date');

        $dataTables = new DataTables;

     	$db_sunrise = \DB::connection('mysql_sunrise_steel_pm');

        $column_alarm = ['LocalTimeDate', 'Tag', 'State', 'Value', 'Type', 'ALMCOMMENT'];
        $column_meter = ['DATETIME', 'Equipment', 'I_Avg', 'VLL_Avg', 'PF', 'KWH1', 'KWH2', 'THDI', 'THDV'];

        $date = ($input['data-source'] == 'data_meter' ? 'DATETIME' : 'LocalTimeDate');
        $column = ($input['data-source'] == 'data_meter' ? $column_meter : $column_alarm);

        if(request()->ajax()){
	        try{
	        	// Formatting form input start-date & end-date
	        	$startDate = date('Y-m-d H:i:s', strtotime($input['start-date']));
	        	$endDate = date('Y-m-d H:i:s', strtotime($input['end-date']));

				// Query Builder
	        	$data = $db_sunrise->table($input['data-source'])->select($column)->whereBetween($date, [$startDate, $endDate ])->orderBy($date);

    	        return $dataTables->query($data)->make(true);  
	        }
	        catch(QueryException $ex){
	        	dd($ex->getMessage()); 
	        }

        }
        else{

			return redirect(route('sunrise.table.index'));

        }

    }
}

</code></pre>

#### frontend javascript: sunrise.js
<pre>
<code>
$(document).ready( function () {

    // When apply filter button is clicked update the table
    $('#apply-filter').click(function(){

        if ( $.fn.dataTable.isDataTable( '#sunrise-table' ) ) {

            // Destroy and empty current table
            $('#sunrise-table').DataTable().destroy();
            $('#sunrise-table').empty();

        }

        switch($('#data-source').val()){
            case 'data_meter':
                $("#sunrise-table").html("<thead><tr><th>Date Time</th><th>Equipment Name</th><th>I_Avg (A)</th><th>V_Avg (V)</th><th>Power Factor</th><th>Acc. Energy Active (GWh)</th><th>Inc. Energy Active (kWh)</th><th>THDi (%)</th><th>THDv (%)</th></tr></thead>");
                break;

            case 'data_alarm':
                $("#sunrise-table").html("<thead><tr><th>Date Time</th><th>Alarm Name</th><th>State</th><th>Value</th><th>Type</th><th>Description</th></tr></thead>");
                break;
        }

        drawTable();

    });

});

function drawTable(){
    var config = {
        processing : true,
        serverSide : true,
        ajax : {
            url: '{{url('/sunrise/table/f')}}',
            type: 'GET',
            data: function(d){
                d.data_source = $('#data-source').val();
                d.start_date = $('#start-date').val();
                d.end_date = $('#end-date').val();                
            },
            success: function(data) {
                console.log('Sunrise table fetch successful');
                //console.log(data);
            },
            error: function (xhr, error, thrown) {
                console.log(table.ajax.params());
                console.log('XHR: ' + xhr + ', error: ' + error + ', thrown: ' + thrown);
            },
            complete: function(data){
                console.log(data);
            }
        },
        language : {
            'zeroRecords' : '0 records to display',
            'emptyTable' : 'No records to display',
        },
    }

    $('#sunrise-table').DataTable(config);
}

</code>
</pre>

#### route: web.php
<pre><code>
Route::get('/sunrise/table', 'SunriseTableController@index')->name('sunrise.table.index'); 
Route::get('/sunrise/table/f', 'SunriseTableController@fetchTableContent')->name('sunrise.table.fetch'); 
</code></pre>

### System details

- Operating System: Windows 10
- PHP Version: 7.3.8
- Laravel Version: 5.8.35
- Laravel-Datatables Version: 9.6.0

#### Result
![image](https://user-images.githubusercontent.com/8023100/64907896-eefe1680-d722-11e9-8339-75e153a6ba96.png)

![image](https://user-images.githubusercontent.com/8023100/64907913-1bb22e00-d723-11e9-8101-1048a8cbfd75.png)

**Comment by lk77** [comment]
Hello,
do you have errors in console ?

**Comment by ronaldramlan** [comment]
No errors

**Comment by ronaldramlan** [comment]
![image](https://user-images.githubusercontent.com/8023100/64908244-a9901800-d727-11e9-87e3-4d62a773eb60.png)

**Comment by DanielGilB** [comment]
Can you check laravel log? It should gives you some info

**Comment by Tiththa** [comment]
Any solution yet? @ronaldramlan

**Comment by ikeliec** [comment]
Hi guys, I am having similar issue. Any update?

**Comment by kodmanyagha** [comment]
This is about version mismatch for `jquery.dataTables.min.js` file. I removed old javascript file from html and added correct version and problem solved.

```
// remove old datatables javascript file
<script src="your/project/folder/whatever/datatable/jquery.dataTables.min.js"></script>

// add this
<script src="https://cdnjs.cloudflare.com/ajax/libs/datatables/1.10.19/js/jquery.dataTables.min.js"></script>

```

Take care of version number. In my project this version number (1.10.19) worked, it may work in your project or not. You must find correct version number.
